cloud/openstack/os_network.py

```diff
@@ -182,10 +193,17 @@ def main():
     provider_physical_network = module.params['provider_physical_network']
     provider_network_type = module.params['provider_network_type']
     provider_segmentation_id = module.params['provider_segmentation_id']
+    project = module.params.pop('project')
 
     try:
         cloud = shade.openstack_cloud(**module.params)
-        net = cloud.get_network(name)
+        if project is not None:
+            project_id = cloud.get_project(project)['id']
```

Comment:
This is going to throw an exception if the project name/id is invalid and you attempt to get the 'id' key for a None value. You should fail_json() if the project is invalid here.